src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs

```diff
@@ -2604,6 +2604,141 @@ public void LastUpdateWins()
+            Dictionary<string, string> expectedUpdatej = new Dictionary<string, string>
+            {
+                {"m1", "second" }
+            };
+
+            ObjectModelHelpers.AssertItemHasMetadata(expectedUpdatei, items[0]);
+            ObjectModelHelpers.AssertItemHasMetadata(expectedUpdatej, items[1]);
+        }
+
+        [Fact]
+        public void UpdatingIndividualItemsProceedsInOrder()
+        {
+            string contents = @"
+<i Include='a;b;c'>
+    <m1>m1_contents</m1>
+</i>
+<i Update='a'>
+    <m1>second</m1>
+</i>
+<i Update='b'>
+    <m1>third</m1>
+</i>
+<i Update='c'>
+    <m1>fourth</m1>
+</i>
+<afterFirst Include='@(i)' />
+<i Update='*'>
+    <m1>sixth</m1>
+</i>
+<afterSecond Include='@(i)' />
+<i Update='b'>
+    <m1>seventh</m1>
+</i>
+<afterThird Include='@(i)' />
+<i Update='c'>
+    <m1>eighth</m1>
+</i>
+<afterFourth Include='@(i)' />
+";
+            IList<ProjectItem> items = ObjectModelHelpers.GetItemsFromFragment(contents, allItems: true);
+            Dictionary<string, string> expectedAfterFirsta = new Dictionary<string, string>
+            {
+                {"m1", "second" }
+            };
+            Dictionary<string, string> expectedAfterFirstb = new Dictionary<string, string>
+            {
+                {"m1", "third" }
+            };
+            Dictionary<string, string> expectedAfterFirstc = new Dictionary<string, string>
+            {
+                {"m1", "fourth" }
+            };
+            Dictionary<string, string> expectedAfterSeconda = new Dictionary<string, string>
+            {
+                {"m1", "sixth" }
+            };
+            Dictionary<string, string> expectedAfterSecondb = new Dictionary<string, string>
+            {
+                {"m1", "sixth" }
+            };
+            Dictionary<string, string> expectedAfterSecondc = new Dictionary<string, string>
+            {
+                {"m1", "sixth" }
+            };
+            Dictionary<string, string> expectedAfterThirda = new Dictionary<string, string>
+            {
+                {"m1", "sixth" }
+            };
+            Dictionary<string, string> expectedAfterThirdb = new Dictionary<string, string>
+            {
+                {"m1", "seventh" }
+            };
+            Dictionary<string, string> expectedAfterThirdc = new Dictionary<string, string>
+            {
+                {"m1", "sixth" }
+            };
+            Dictionary<string, string> expectedAfterFourtha = new Dictionary<string, string>
+            {
+                {"m1", "sixth" }
+            };
+            Dictionary<string, string> expectedAfterFourthb = new Dictionary<string, string>
+            {
+                {"m1", "seventh" }
+            };
+            Dictionary<string, string> expectedAfterFourthc = new Dictionary<string, string>
+            {
+                {"m1", "eighth" }
+            };
+
+            ObjectModelHelpers.AssertItemHasMetadata(expectedAfterFirsta, items[3]);
```

Comment:
Should AssertItemHasMetadata have an overload that accepts a tuple name-value, so you can eliminate all these dictionaries in the test?